### ISSUE TITLE
Wekan dont Save Language

### ISSUE THREAD

**Issue by EDV-Muencheberg** [wekan/wekan#3629]
Hello,

if you cahnge the Language to german or something else and reopen the Browser the Language changed back to the default (english)

Regards

**Comment by xet7** [comment]
@EDV-Muencheberg 

What version of Wekan you have?

**Comment by xet7** [comment]
@EDV-Muencheberg 

With newest Wekan, when I reload webbrowser page, after webpage is loaded it still has the same language I selected, in my case Finnish. While loading webpage, for a small time it shows English, because all translations have not been loaded yet, but after all translations have been loaded and Javascript code has run, it is still in Finnish.

**Comment by xet7** [comment]
@EDV-Muencheberg

I changed to Deutch, and after reloading webpage, it is still Deutch.

**Comment by xet7** [comment]
If still after updating to newest Wekan this happens, please comment on this issue, so I will then open this issue again.

**Comment by fcoufour** [comment]
@xet7 We found something strange relative to how user language is handle.

Without any language in user mongodb document:
- english (❌french expected) after login when french (France) is selected in Chrome settings, english (❌french expected) after refresh, english (❌french expected) selected in language menu
- english (❌french expected) after login when french is selected in Chrome settings, french (✅ok) after refresh, french (✅ok) selected in language menu

Without "de" language in user mongodb document:
- english (❌german expected) after login when french (France) is selected in Chrome settings, german (✅ok) after refresh, german (✅ok) selected in language menu
- german (✅ok) after login when french is selected in Chrome settings, german (✅ok) after refresh, german (✅ok) selected in language menu 

👉 the country language code (like fr-FR) seems to be ignored
👉 ths behaviour is not always as expected

Are you aware of this?

Tested with current last version of Chrome and with last version of Wekan (5.14.0)

**Comment by jimmy09140456** [comment]
Same here with later version of Chrome and Wekan 5.14.

1. Wekan showed Traditional Chinese by default which is my default language in Google Chrome
2. I manually changed the language to English
3. Refresh, it was still showing English
4. Using Wekan for a while (about 1 hour?)
5. The UI was partially showing Traditional Chinese
6. Refresh, the entire UI became Traditional Chinese

**Comment by xet7** [comment]
For any translations that is only partially translated, please check that all are translated at:
https://www.transifex.com/wekan/wekan

**Comment by xet7** [comment]
Very often someone requests some language duplicate.
For example:
- Swedish (Sweden) is same as Swedish
- Chinese (China) is same as Chinese
- etc

So if there exists for example already either Language (Country) or Language, I have asked for confirmation is it different, and not added duplicate.

In addition of this bug, there are [other translation bugs](https://github.com/wekan/wekan/labels/Translations)

**Comment by StefanNeubert** [comment]
Same here with 5.17. Reloading page or closing browser the language changes back to english.
Running on docker.
![wekan_language](https://user-images.githubusercontent.com/66677532/113250579-0e3aa580-92c1-11eb-9544-0aea11fdbf32.gif)

**Comment by EDV-Muencheberg** [comment]
Same Problem with version 5.15.0 with snap.

**Comment by StefanNeubert** [comment]
@EDV-Muencheberg are you running on a sublocation? Maybe this is an issue an related to #3384

**Comment by EDV-Muencheberg** [comment]
@StefanNeubert yes we run it on a sublocation.

**Comment by Meeques** [comment]
I've got this problem, too. Our WeKan instance is always the newest version (unbuntu snap). In Firefox, I don't have this issue, only in Chrome (Version 97.0.4692.99). A colleague of mine has the same problem. When reloading the page, it switches from german to english.

**Comment by xet7** [comment]
@Meeques 

Try:

1. Clear Chrome history, everything but not cookies
2. Close Chrome
3. Open Chrome
4. Open WeKan board page
5. Reload page
6. Does it stay German?

**Comment by Meeques** [comment]
@xet7 Unfortunately the problem persists. 
After doing all that (twice) WeKan still switches to english.

**Comment by jimmy09140456** [comment]
How is your Wekan instance running? Under a dedicated domain or sub-url? I found out that switching language won't work if your Wekan is running under sub-url. Not only this, many other features, such as create shortcut on mobile, are also not working. There are some static files not being resolved under the sub-url and the Wekan front-end still trying to fetch them from the root path/

**Comment by Meeques** [comment]
It is a dedicated domain. But the browser console does cast some errors. I will ask the data center for help. thank you!

**Comment by xet7** [comment]
@Meeques 

If there are websockets or other errors, check webserver config. Examples for various webservers at right menu of page:
https://github.com/wekan/wekan/wiki

For example, there are Caddy, Nginx, Apache etc examples.

**Comment by xet7** [comment]
@Meeques 

What errors there are at browser inspect console? Please copy them to text editor, anonymize it like replace your domains with example.com etc, and copy paste those errors to comment to this issue, if possible. Thanks!

**Comment by Meeques** [comment]
@xet7 Yes, there are websocket errors, I will check in with the server administrator. Thanks for the link.

Here is the requested txt file:
[www.uni-example.de-1643104186443.txt](https://github.com/wekan/wekan/files/7932496/www.uni-example.de-1643104186443.txt)

**Comment by xet7** [comment]
@Meeques 

Current WeKan requires that websockets are configured to work.

In my install of WeKan using Caddy webserver, I don't have any of those errors.
